frontend/src/components/ForumPost.tsx

```diff
@@ -1,41 +1,79 @@
 import { Text, Card, Spacer, Image } from '@geist-ui/react';
 import { Link } from "react-router-dom";
 
-interface Post {
-    id: number,
-    user: number,
-    status: string,
-    date_posted: Date,
-    title: string,
+// interface Post {
+//     id: number,
+//     user: number,
+//     status: string,
+//     date_posted: Date,
+//     title: string,
+//     content: string,
+//     messages: number[]
+// }
+
+// interface PostItemProps {
+//     post: Post
+// }
+
+interface User {
+    id: string,
+    username: string,
+    image: string,
+    orders: string[]
+}
+
+interface ForumPost {
+    authorId: string
+    datePosted: Date
+    content: string
+  }
+  
+
+interface ForumThread {
+    id: string,
+    authorId: string,
+    datePosted: Date,
     content: string,
-    messages: number[]
+    title: string,
+    status: string,
+    posts: ForumPost[]
 }
 
 interface PostItemProps {
-    post: Post
+    post: ForumThread
 }
 
 
-
 const ForumPost = (props: PostItemProps) => {
   const { post } = props;
 
-  function getUserName(){
-    if(post.user === 1){
-        return "Max Dunaevschi"
-    }
-    else if(post.user === 2){
-        return "Gabriel Magendzo"
-    }
-    else if(post.user === 3){
-        return "Oscar Kav"
-    }
-    else if(post.user === 4){
-        return "Achilles Ecos"
-    }
-    else {
-        return "Gialon Kasha"
+  async function fetchUser(){
+    try {
+        const response = await fetch(`http://localhost:5001/claymore-d6749/us-central1/default/user?userId=${post.authorId}`).then((res) => (res.json()));
```

Comment:
Hardcoding the localhost path seems bad because when we deploy we need to change it to the actual path. I suggest to make it a global variable or something and share it across the frontent

Reply:
Max has a fix in his previous PR to use a config, can just adopt that here.